HW0/P5/P5.txt

```diff
@@ -1 +1,54 @@
 # Your answers here.
+
+1. It would take 255 or 256 seconds (depending on whether or not you count
+adding the first value to 0 as a second or not) 
+
+2. We have that 256 / 8 = 32, so in just 32 seconds, each of the 8 workers 
+will have a number that represents the number of pennies they counted
+indivudually. Then, assuming that communication takes no time, in 1 second we 
+could take the 8 numbers and make them into 4, by assigning 4 people 2 numbers 
+to add each. Then, in another second, we could turn 4 numbers into 2, and 
+then 2 numbers into 1. Therefore, in total, this process takes 35 seconds. 
+
+3. Assuming that it doesn't take any time just to type a number into the 
+calculator, every second we will cut the number of numbers left to add in 
+half. Therefore, we will finish in log_2(n) steps, where n is the number of 
+bags. In our case, log_2(256) = 8. 
+
+4. As N grows large, we see that the amount of time for one person to 
+count grows linearly, while if parallelized with infinite number of cashiers
+grows logarithmically. See graph, where the red line is non-parallelized 
+and the green line is. 
```

Comment:
Got a similar graph. I think it might be helpful to set your y axis to a log scale in order to see what's really going on there.